Sources/ImperialCore/Helpers/String+Tools.swift

```diff
@@ -2,16 +2,31 @@ import Foundation
 import RoutingKit
 
 extension String {    
```

Comment:
Change pathComponents var to a func.
Break url path into components like before.
Remove any duplication starting with first index. Stop at the first index that does not match.